tests/basic/volumeplacementstrategy_test.go

```diff
@@ -0,0 +1,243 @@
+
+import (
+	"fmt"
+
+	. "github.com/onsi/ginkgo"
+	"github.com/portworx/sched-ops/k8s/talisman"
+	"github.com/portworx/talisman/pkg/apis/portworx/v1beta1"
+	"github.com/portworx/talisman/pkg/apis/portworx/v1beta2"
+	"github.com/portworx/torpedo/drivers/scheduler"
+	"github.com/portworx/torpedo/pkg/log"
+	"github.com/portworx/torpedo/pkg/testrailuttils"
+	"github.com/portworx/torpedo/pkg/vpsutil"
+	. "github.com/portworx/torpedo/tests"
+)
+
+var _ = Describe("{VolumePlacementStrategyFunctional}", func() {
+	var testrailID, runID int
+	var contexts []*scheduler.Context
+	var namespacePrefix string
+
+	JustBeforeEach(func() {
+		runID = testrailuttils.AddRunsToMilestone(testrailID)
+
+		StartTorpedoTest("VolumePlacementStrategyFunctional", "Functional Tests for VPS", nil, testrailID)
+	})
+
+	Context("VolumePlacementStrategyValidation", func() {
+		var vpsTestCase VolumePlaceMentStrategyTestCase
+
+		testValidateVPS := func() {
+			It("has to deploy VPS and validate the scheduled application follow specified rules", func() {
+				Step("Deploying VPS", func() {
+					log.InfoD("Deploy VPS for %v", vpsTestCase.TestName())
+					err := vpsTestCase.DeployVPS()
+					log.FailOnError(err, "Failed to Deploy VPS Spec")
+				})
+
+				Step("Deploy and Validate Applications", func() {
+					log.InfoD("Deploy Applications")
+					contexts = make([]*scheduler.Context, 0)
+					for i := 0; i < Inst().GlobalScaleFactor; i++ {
+						contexts = append(contexts, ScheduleApplications(fmt.Sprintf("%s-%d", namespacePrefix, i))...)
+					}
+					log.InfoD("Validate Applications")
+					ValidateApplications(contexts)
+				})
+
+				Step("Validate Deployment with VPS", func() {
+					err := vpsTestCase.ValidateVPSDeployment(contexts)
+					log.FailOnError(err, "Failed to Validate Deployments with respect to VPS")
+				})
+
+				Step("Destroy VPS Deployment", func() {
+					err := vpsTestCase.DestroyVPSDeployment()
+					log.FailOnError(err, "Failed to Destroy VPS Deployments")
+				})
+
+			})
+		}
+
+		// test mongo volume anti affinity
+		Context("{VPSMongoAntiAffinity}", func() {
+			BeforeEach(func() {
+				namespacePrefix = "mongovpsantiaffinity"
+				vpsTestCase = &mongoVPSAntiAffinity{}
+			})
+			testValidateVPS()
+		})
+
+		// test mongo volume anti affinity
+		Context("{VPSMongoAffinity}", func() {
+			BeforeEach(func() {
+				namespacePrefix = "mongovpsaffinity"
+				vpsTestCase = &mongoVPSAffinity{}
+			})
+			testValidateVPS()
+		})
+	})
+
+	AfterEach(func() {
+		Step("destroy apps", func() {
+			log.InfoD("destroying apps")
+			if CurrentGinkgoTestDescription().Failed {
+				log.InfoD("not destroying apps because the test failed\n")
+				return
+			}
+			for _, ctx := range contexts {
+				TearDownContext(ctx, map[string]bool{scheduler.OptionsWaitForResourceLeakCleanup: true})
+			}
+
+		})
+	})
+
+	AfterEach(func() {
+		AfterEachTest(contexts, testrailID, runID)
+		defer EndTorpedoTest()
+	})
+})
+
+type VolumePlaceMentStrategyTestCase interface {
```

Comment:
can you move this and other non tests functions to new file under vpsutil. so that we can use in other tests if required